api/v1alpha1/hostedcluster_types.go

```diff
@@ -14,8 +14,10 @@ func init() {
 
 const (
 	// AuditWebhookKubeconfigKey is the key name in the AuditWebhook secret that stores audit webhook kubeconfig
-	AuditWebhookKubeconfigKey          = "webhook-kubeconfig"
-	DisablePKIReconciliationAnnotation = "hypershift.openshift.io/disable-pki-reconciliation"
+	AuditWebhookKubeconfigKey                 = "webhook-kubeconfig"
+	DisablePKIReconciliationAnnotation        = "hypershift.openshift.io/disable-pki-reconciliation"
+	IdentityProviderOverridesAnnotationPrefix = "idpoverrides.hypershift.openshift.io/"
+	OauthLoginURLOverrideAnnotation           = "oauth.hypershift.openshift.io/login-url-override"
```

Comment:
this might seem unecessary at first however it's important due to the serving cert we set up:
We need the login url to be the external url that has TLS certs issued by our top level CA provider (Digicert) so it validates in a browser. 

However with the current wiring: the endpoint points to the *internal* cluster endpoint that uses the in cluster PKI infrastructure. We can't just switch to the external endpoint for everything either because then when the kubeconfig gets wired with the CA the TLS validation fails of the kubeconfig that goes in the hostedclusters namespace. 

This provides a low overhead for us just to set the login url to the external serving cert endpoint and then all requirements are met :)